src/components/radio/RadioSelect.tsx

```diff
@@ -0,0 +1,38 @@
+import * as React from 'react';
+import { ValidComponentChildren } from '../../utils/ValidComponentChildren';
+import { createChainedFunction } from '../../utils/createChainedFunction';
+
+export interface IRadioSelectProps {
+  name?: string;
+  value?: string;
+  disabled?: boolean;
+  onChange?: (value: string) => void;
+}
+
+export class RadioSelect extends React.Component<IRadioSelectProps, any> {
+  handleToggle(value: string) {
+    if (this.props.onChange) {
+      this.props.onChange(value);
+    }
+  }
+
+  render() {
+    return (
+      <div className='form-control radio-select'>
+        {
+          ValidComponentChildren.map(this.props.children, (child: React.ReactElement<any>) => {
+            const { value, onChange } = child.props;
+            const handler = () => this.handleToggle(value);
+
+            return React.cloneElement(child, {
+              name: child.props.name || this.props.name,
+              checked: this.props.value === value,
+              disabled: this.props.disabled,
+              onChange: createChainedFunction(onChange, handler),
```

Comment:
I don't understand why you need to chain here. You override the prop `onChange` of your children to call handler? Why not simply 

```
onChange: () => {
    child.props.onChange && child.props.onChange(); 
    this.handleToggle(child.props.value)
}
```